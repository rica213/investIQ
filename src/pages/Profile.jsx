import React from 'react';
import styles from '../styles/Profile.module.css';

const Profile = () => (
  <article className={styles.profile}>
    <section className={styles.headline}>
      <img src="" alt="logo goes here" className={styles.logo} />
      <div>
        <h2 className={styles.name}>Apple</h2>
        <h3 className={styles.symbol}>AAPL</h3>
      </div>
    </section>
    <div className={styles.title}>Profile details</div>
    <section className={styles.details}>
      <div className={styles.row}>
        <span>Price:</span>
        <p>some metrics</p>
      </div>
      <div className={styles.row}>
        <span>Currency:</span>
        <p>some metrics</p>
      </div>
      <div className={styles.row}>
        <span>CEO:</span>
        <p>some metrics</p>
      </div>
      <div className={styles.row}>
        <span>Industry:</span>
        <p>some metrics</p>
      </div>
      <div className={styles.row}>
        <span>Website:</span>
        <p>some metrics</p>
      </div>
    </section>
  </article>
);

export default Profile;
