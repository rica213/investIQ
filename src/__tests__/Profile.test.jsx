import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import configureMockStore from 'redux-mock-store';
import Profile from '../pages/Profile';

const mockStore = configureMockStore([]);

describe('Profile component', () => {
  it('should match snapshot', () => {
    const store = mockStore({
      profile: {
        details: [
          {
            companyName: 'Apple Inc.',
            symbol: 'AAPL',
            image: 'https://example.com/apple.png',
            price: '124.97',
            currency: 'USD',
            ceo: 'Tim Cook',
            industry: 'Technology',
            website: 'https://www.apple.com/',
          },
        ],
        isLoading: false,
      },
    });

    const { container } = render(
      <Provider store={store}>
        <Profile />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
