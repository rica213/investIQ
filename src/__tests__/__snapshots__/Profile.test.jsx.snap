// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Profile component > should match snapshot 1`] = `
<div>
  <article
    class="_profile_331855"
  >
    <section
      class="_headline_331855"
    >
      <img
        alt="Apple Inc."
        class="_logo_331855"
        src="https://example.com/apple.png"
      />
      <div>
        <h2
          class="_name_331855"
        >
          Apple Inc.
        </h2>
        <h3
          class="_symbol_331855"
        >
          AAPL
        </h3>
      </div>
    </section>
    <article
      class="_summary_78e985"
    >
      <p>
        Profile details
      </p>
    </article>
    <section
      class="_details_331855"
    >
      <div
        class="_row_331855"
      >
        <span>
          Price:
        </span>
        <p>
          124.97
        </p>
      </div>
      <div
        class="_row_331855"
      >
        <span>
          Currency:
        </span>
        <p>
          USD
        </p>
      </div>
      <div
        class="_row_331855"
      >
        <span>
          CEO:
        </span>
        <p>
          Tim Cook
        </p>
      </div>
      <div
        class="_row_331855"
      >
        <span>
          Industry:
        </span>
        <p>
          Technology
        </p>
      </div>
      <div
        class="_row_331855"
      >
        <span>
          Website:
        </span>
        <p>
          https://www.apple.com/
        </p>
      </div>
    </section>
  </article>
</div>
`;
